opentelemetry-api/src/opentelemetry/metrics/meter.py

```diff
@@ -0,0 +1,210 @@
+    def __init__(self, value, **attributes: Attributes):
+        super().__init__(value, **attributes)
+
+
+class Meter(ABC):
+
+    # FIXME make unit and description be "" if unit or description are None
+    @abstractmethod
+    def create_counter(self, name, unit="", description="") -> Counter:
+        pass
+
+    @abstractmethod
+    def create_up_down_counter(
+        self, name, unit="", description=""
+    ) -> UpDownCounter:
+        pass
+
+    @abstractmethod
+    def create_observable_counter(
+        self, name, callback, unit="", description=""
+    ) -> ObservableCounter:
+        pass
+
+    @abstractmethod
+    def create_histogram(self, name, unit="", description="") -> Histogram:
+        pass
+
+    @abstractmethod
+    def create_observable_gauge(
+        self, name, callback, unit="", description=""
+    ) -> ObservableGauge:
+        pass
+
+    @abstractmethod
+    def create_observable_up_down_counter(
+        self, name, callback, unit="", description=""
+    ) -> ObservableUpDownCounter:
+        pass
+
+    @staticmethod
+    def check_unique_name(checker):
+        def wrapper_0(method):
+            @wraps(method)
+            def wrapper_1(self, name, unit="", description=""):
+                checker(self, name)
+                return method(self, name, unit=unit, description=description)
+
+            return wrapper_1
+
+        return wrapper_0
+
+
+class DefaultMeter(Meter):
+    def __init__(self):
+        self._instrument_names = set()
+
+    def _instrument_name_checker(self, name):
+
+        if name in self._instrument_names:
+            raise Exception("Instrument name {} has been used already")
+
+        self._instrument_names.add(name)
+
+    @Meter.check_unique_name(_instrument_name_checker)
+    def create_counter(self, name, unit="", description="") -> Counter:
+        return DefaultCounter(name, unit=unit, description=description)
+
+    @Meter.check_unique_name(_instrument_name_checker)
+    def create_up_down_counter(
+        self, name, unit="", description=""
+    ) -> UpDownCounter:
+        return DefaultUpDownCounter(name, unit=unit, description=description)
+
+    @Meter.check_unique_name(_instrument_name_checker)
+    def create_observable_counter(
+        self, name, callback, unit="", description=""
+    ) -> ObservableCounter:
+        return DefaultObservableCounter(
+            name,
+            callback,
+            unit=unit,
+            description=description,
+        )
+
+    @Meter.check_unique_name(_instrument_name_checker)
+    def create_histogram(self, name, unit="", description="") -> Histogram:
+        return DefaultHistogram(name, unit=unit, description=description)
+
+    @Meter.check_unique_name(_instrument_name_checker)
+    def create_observable_gauge(
+        self, name, callback, unit="", description=""
+    ) -> ObservableGauge:
+        return DefaultObservableGauge(  # pylint: disable=abstract-class-instantiated
```

Comment:
Are the Default classes abstract? How come pylint complains about these?

Reply:
That I also want to know :laughing:

Reply:
The pylint comment was not needed, I have removed it.